src/driver/m3_subset_test_point_generator.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates test points for the subset use case with the m3 strategy."""
+
+from typing import Iterable, List
+
+import numpy as np
+from itertools import combinations
+from wfa_planning_evaluation_framework.data_generators.data_set import DataSet
+from wfa_planning_evaluation_framework.driver.test_point_generator import (
+    TestPointGenerator,
+)
+
+
+class M3SubsetTestPointGenerator(TestPointGenerator):
+    """Generates test points for the subset use case with the m3 strategy.
+
+    In M3, the training points are the frequency histograms of
+    - each of the single publishers,
+    - each of the all-but-one subsets of publishers,
```

Comment:
If I have never worked on this, I would be stuck at understanding the meaning of "the frequency histogram of an all-but-one subset".

Reply:
Thanks for letting me know.  Which part do you think is difficult to understand?  The meaning of "all-but-one subset", or how to calculate the frequency histogram of it?

Reply:
I mean its meaning. On the other hand, I think it is very minor after a second thought. :)